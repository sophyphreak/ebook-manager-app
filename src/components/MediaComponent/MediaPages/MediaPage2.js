import React, { Component } from 'react';
import { Card, CardHeader, CardBody, Button, Form } from 'reactstrap';

import Price from '../../FormElements/Price';
import DatesPicker from '../../FormElements/DatesPicker';

const MediaPage2 = ({
  error,
  onBack,
  onSubmit,

  regPrice,
  onRegPriceChange,

  salePrice,
  onSalePriceChange,

  startDate,
  endDate,
  focusedInput,
  onDatesChange,
  onFocusChange
}) => (
  <Card className="animated fadeIn">
    <CardHeader>
      <h4>
        <strong>Social Media Submission - 2 of 3</strong>
      </h4>
    </CardHeader>
    <CardBody>
      <Form onSubmit={onSubmit}>
        <Price
          label="Regular Price*"
          price={regPrice}
          onPriceChange={onRegPriceChange}
          hasError={!!error.message}
          errorMessage={error.regPrice}
        />
        <Price
          label="Sale Price (if applicable)"
          price={salePrice}
          onPriceChange={onSalePriceChange}
          hasError={!!error.message}
          errorMessage={error.salePrice}
        />
        <DatesPicker
          label="Dates for promotion (if applicable)"
          startDate={startDate}
          endDate={endDate}
          onDatesChange={onDatesChange}
          focusedInput={focusedInput}
          onFocusChange={onFocusChange}
        />
        {error.message && <p>{error.message}</p>}
        <Button
          className="back-button"
          onClick={onBack}
          size="sm"
          color="basic"
        >
          Go Back
        </Button>
        <Button type="submit" size="sm" color="primary" value="Submit">
          Next Section
        </Button>
      </Form>
    </CardBody>
  </Card>
);

export default MediaPage2;
