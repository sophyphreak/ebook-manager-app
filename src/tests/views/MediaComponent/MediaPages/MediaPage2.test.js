import React from 'react';
import { shallow } from 'enzyme';
import MediaPage2 from '../../../../views/MediaView/MediaPages/MediaPage2';

let error, wrapper;

beforeEach(() => {
  error = {
    message: ''
  };
  wrapper = shallow(<MediaPage2 error={error} />);
});

test('should render MediaPage2 correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
