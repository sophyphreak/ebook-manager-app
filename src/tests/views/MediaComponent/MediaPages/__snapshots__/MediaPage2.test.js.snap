// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MediaPage2 correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MediaPage2
    error={
      Object {
        "message": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <CardHeader
          tag="div"
        >
          <h4>
            <strong>
              Social Media Submission - 2 of 3
            </strong>
          </h4>
        </CardHeader>,
        <CardBody
          tag="div"
        >
          <Form
            tag="form"
          >
            <Price
              hasError={false}
              label="Regular Price*"
            />
            <Price
              hasError={false}
              label="Sale Price (if applicable)"
            />
            <DatesPicker
              label="Dates for promotion (if applicable)"
            />
            
            <Button
              className="back-button"
              color="basic"
              size="sm"
              tag="button"
            >
              Go Back
            </Button>
            <Button
              color="primary"
              size="sm"
              tag="button"
              type="submit"
              value="Submit"
            >
              Next Section
            </Button>
          </Form>
        </CardBody>,
      ],
      "className": "animated fadeIn",
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <h4>
            <strong>
              Social Media Submission - 2 of 3
            </strong>
          </h4>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <strong>
              Social Media Submission - 2 of 3
            </strong>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Social Media Submission - 2 of 3",
            },
            "ref": null,
            "rendered": "Social Media Submission - 2 of 3",
            "type": "strong",
          },
          "type": "h4",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Form
            tag="form"
          >
            <Price
              hasError={false}
              label="Regular Price*"
            />
            <Price
              hasError={false}
              label="Sale Price (if applicable)"
            />
            <DatesPicker
              label="Dates for promotion (if applicable)"
            />
            
            <Button
              className="back-button"
              color="basic"
              size="sm"
              tag="button"
            >
              Go Back
            </Button>
            <Button
              color="primary"
              size="sm"
              tag="button"
              type="submit"
              value="Submit"
            >
              Next Section
            </Button>
          </Form>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Price
                hasError={false}
                label="Regular Price*"
              />,
              <Price
                hasError={false}
                label="Sale Price (if applicable)"
              />,
              <DatesPicker
                label="Dates for promotion (if applicable)"
              />,
              "",
              <Button
                className="back-button"
                color="basic"
                size="sm"
                tag="button"
              >
                Go Back
              </Button>,
              <Button
                color="primary"
                size="sm"
                tag="button"
                type="submit"
                value="Submit"
              >
                Next Section
              </Button>,
            ],
            "onSubmit": undefined,
            "tag": "form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "errorMessage": undefined,
                "hasError": false,
                "label": "Regular Price*",
                "onPriceChange": undefined,
                "price": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "errorMessage": undefined,
                "hasError": false,
                "label": "Sale Price (if applicable)",
                "onPriceChange": undefined,
                "price": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "endDate": undefined,
                "focusedInput": undefined,
                "label": "Dates for promotion (if applicable)",
                "onDatesChange": undefined,
                "onFocusChange": undefined,
                "startDate": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Go Back",
                "className": "back-button",
                "color": "basic",
                "onClick": undefined,
                "size": "sm",
                "tag": "button",
              },
              "ref": null,
              "rendered": "Go Back",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Next Section",
                "color": "primary",
                "size": "sm",
                "tag": "button",
                "type": "submit",
                "value": "Submit",
              },
              "ref": null,
              "rendered": "Next Section",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <CardHeader
            tag="div"
          >
            <h4>
              <strong>
                Social Media Submission - 2 of 3
              </strong>
            </h4>
          </CardHeader>,
          <CardBody
            tag="div"
          >
            <Form
              tag="form"
            >
              <Price
                hasError={false}
                label="Regular Price*"
              />
              <Price
                hasError={false}
                label="Sale Price (if applicable)"
              />
              <DatesPicker
                label="Dates for promotion (if applicable)"
              />
              
              <Button
                className="back-button"
                color="basic"
                size="sm"
                tag="button"
              >
                Go Back
              </Button>
              <Button
                color="primary"
                size="sm"
                tag="button"
                type="submit"
                value="Submit"
              >
                Next Section
              </Button>
            </Form>
          </CardBody>,
        ],
        "className": "animated fadeIn",
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <h4>
              <strong>
                Social Media Submission - 2 of 3
              </strong>
            </h4>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <strong>
                Social Media Submission - 2 of 3
              </strong>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Social Media Submission - 2 of 3",
              },
              "ref": null,
              "rendered": "Social Media Submission - 2 of 3",
              "type": "strong",
            },
            "type": "h4",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Form
              tag="form"
            >
              <Price
                hasError={false}
                label="Regular Price*"
              />
              <Price
                hasError={false}
                label="Sale Price (if applicable)"
              />
              <DatesPicker
                label="Dates for promotion (if applicable)"
              />
              
              <Button
                className="back-button"
                color="basic"
                size="sm"
                tag="button"
              >
                Go Back
              </Button>
              <Button
                color="primary"
                size="sm"
                tag="button"
                type="submit"
                value="Submit"
              >
                Next Section
              </Button>
            </Form>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Price
                  hasError={false}
                  label="Regular Price*"
                />,
                <Price
                  hasError={false}
                  label="Sale Price (if applicable)"
                />,
                <DatesPicker
                  label="Dates for promotion (if applicable)"
                />,
                "",
                <Button
                  className="back-button"
                  color="basic"
                  size="sm"
                  tag="button"
                >
                  Go Back
                </Button>,
                <Button
                  color="primary"
                  size="sm"
                  tag="button"
                  type="submit"
                  value="Submit"
                >
                  Next Section
                </Button>,
              ],
              "onSubmit": undefined,
              "tag": "form",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "errorMessage": undefined,
                  "hasError": false,
                  "label": "Regular Price*",
                  "onPriceChange": undefined,
                  "price": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "errorMessage": undefined,
                  "hasError": false,
                  "label": "Sale Price (if applicable)",
                  "onPriceChange": undefined,
                  "price": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "endDate": undefined,
                  "focusedInput": undefined,
                  "label": "Dates for promotion (if applicable)",
                  "onDatesChange": undefined,
                  "onFocusChange": undefined,
                  "startDate": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Go Back",
                  "className": "back-button",
                  "color": "basic",
                  "onClick": undefined,
                  "size": "sm",
                  "tag": "button",
                },
                "ref": null,
                "rendered": "Go Back",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Next Section",
                  "color": "primary",
                  "size": "sm",
                  "tag": "button",
                  "type": "submit",
                  "value": "Submit",
                },
                "ref": null,
                "rendered": "Next Section",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
